typegen/generate_typing.py

```diff
@@ -0,0 +1,168 @@
+from pathlib import Path
+from typing import Final
+
+
+def write_annotations(output_folder: Path, strict: bool) -> None:
+    """
+    Generate and write typing annotations for the library
+
+    Parameters
+    ----------
+    output_folder : Path
+        path within a package that gets updated
+
+    strict : bool
+        whether to generate strict types or allow conversions
+
+    Returns
+    -------
+    None
+    """
+    generate_template = _generate_strict_template if strict else _generate_union_template
+    for dimensions in _DIMENSION_TYPES:
+        contents = "\n".join([_annotate_type(dimensions, type_name, strict) for type_name in _DATA_TYPES])
```

Comment:
No, need to use list-comprehension inside join, the iterator is enough.

Solution: Just remove the square brackets.